pkg/generate/app/app.go

```diff
@@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
 		r.Name = name
 	}
 
-	selector := map[string]string{
-		"deploymentconfig": r.Name,
+	var selector map[string]string
+	if len(r.Labels) > 0 {
```

Comment:
If you have a DC with no selector or labels set, does it inherit the DCs
labels?

On Jul 23, 2015, at 3:39 PM, Cesar Wong <notifications@github.com> wrote:

In pkg/generate/app/app.go
<https://github.com/openshift/origin/pull/3882#discussion_r35362372>:

> @@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
>  		r.Name = name
>  	}
>
> -	selector := map[string]string{
> -		"deploymentconfig": r.Name,
> +	var selector map[string]string
> +	if len(r.Labels) > 0 {

 Do we support selector inheritance as I described? Inherit the labels by
default?

I understood this to mean that we apply use the labels in the selector, is
there more to it than that? (we still apply the labels to all objects
generated)

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3882/files#r35362372>.